File: HorseRacing.Web/src/interfaces/api/contracts/model/auth/requests/login/login-request.ts
import type { BaseRequest } from "~/interfaces/api/contracts/base/requests/base-request";
import type { LoginRequestDto } from "./login-request-dto";

export class LoginRequest implements BaseRequest<LoginRequestDto> {
    Data: LoginRequestDto;

    constructor(data?: LoginRequestDto) {
        this.Data = data || { UserName: '', Password: '' };
    }
}

File: HorseRacing.Contracts/Models/Authentication/Requests/Login/LoginRequest.cs
using HorseRacing.Contracts.Base.Requests;

namespace HorseRacing.Contracts.Models.Authentication.Requests.Login
{
    /// <summary>
    /// Модель запроса для авторизации
    /// </summary>
    public class LoginRequest : BaseRequest<LoginRequestDto>
    {
        public LoginRequest() : base() { }
        public LoginRequest(LoginRequestDto data) : base(data) { }
    }
}
